synfig-studio/src/gui/render.cpp

```diff
@@ -339,39 +371,104 @@ RenderSettings::on_render_pressed()
 		catch(std::runtime_error& x)
 		{
 			canvas_interface_->get_ui_interface()->error(_("Unable to determine proper target from filename."));
-			return;
+			return false;
 		}
 	}
 
 	if(filename.empty() && calculated_target_name!="null")
 	{
 		canvas_interface_->get_ui_interface()->error(_("A filename is required for this target"));
-		return;
+		return false;
 	}
-
-	hide();
-
-	render_passes.clear();
-	if (toggle_extract_alpha.get_active())
+	
+	String full_name(filename); //include possible sequence_separator + time for image sequences
+	String extension(filename_extension(filename));
+	bool ext_multi_file = false; //output target is an image sequence
+	int n_frames_overwrite = 0;
+	
+	//Retrieve current render settings
+	RendDesc rend_desc(widget_rend_desc.get_rend_desc());
+	
+	//Check format which could have an image sequence as output
+	//If format is selected in comboboxtext_target
+	std::map<std::string,std::string> ext_multi = {{"bmp",".bmp"},{"cairo_png",".png"},
+					{"imagemagick",".png"}, {"jpeg",".jpg"},{"mng",".mng"},
+					{"openexr",".exr"},{"png",".png"},{"ppm",".ppm"}};
+	
+	std::map<std::string,std::string>::iterator ext_multi_it;
+	ext_multi_it = ext_multi.find(calculated_target_name);
+	
+	//calculated_target_name is a candidate with known output target (not Auto)
+	if(ext_multi_it != ext_multi.end())
 	{
-		String filename_alpha(filename_sans_extension(filename)+"-alpha"+filename_extension(filename));
-
-		render_passes.push_back(make_pair(TARGET_ALPHA_MODE_EXTRACT, filename_alpha));
-		render_passes.push_back(make_pair(TARGET_ALPHA_MODE_REDUCE, filename));
+		extension = ext_multi_it->second;
+		ext_multi_file = true;
+	}
+	//otherwise Auto is selected
+	else
+	{
+		std::list<std::string> ext_multi_auto = {{".bmp"}, {".png"},
+					{".jpg"},{".exr"},{".ppm"}};
+	
+		ext_multi_file = (find(ext_multi_auto.begin(), ext_multi_auto.end(),
+				filename_extension(filename)) != ext_multi_auto.end());
+	}
 
-	} else {
-		render_passes.push_back(make_pair(TARGET_ALPHA_MODE_KEEP, filename));
+	//Image sequence: filename + sequence_separator + time
+	if(!toggle_single_frame.get_active() && ext_multi_file &&
```

Comment:
I wonder if it would be better to place this `if` statement before investigate if it is a multiple-files target.
`if(!toggle_single_frame.get_active() && (rend_desc.get_frame_end() - rend_desc.get_frame_start()) > 0)`

Reply:
I think I can simplify it a bit and move the condition to avoid useless checks.